resources/js/components/Search/Autocomplete.vue

```diff
@@ -29,6 +29,27 @@ export default {
         this.$nextTick(() => {
             requestAnimationFrame(() => this.$emit('mounted'))
         })
+
+        const stateChanged = useDebounceFn((event) => {
+            const query = event?.payload?.query
+
+            if (!query) {
+                return
+            }
+
+            rapidezAPI('post', '/search', {
+                q: query,
+                results: event?.payload?.nbHits,
+            })
```

Comment:
My only worry here is that it makes abuse of this endpoint extremely easy. Should there be some sort of safeguarding on here?

Reply:
There is not really a foolproof way to prevent abuse here.
But with the latest commit we can significantly reduce abuse.

We now check the CSRF token, so the requests must come from customers who have a currently active session on the site.
It can't completely prevent abuse but it will make it more difficult.